src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -1004,7 +1004,17 @@ export class ChatWidget extends Disposable implements IChatWidget {
 					`${query.prefix} ${editorValue}`;
 			const isUserQuery = !query || 'prefix' in query;
 
-
+			const { promptInstructions } = this.inputPart.attachmentModel;
+			const instructionsEnabled = promptInstructions.featureEnabled;
+			if (instructionsEnabled) {
+				// instruction files may have nested child references to other prompt
+				// files that are resolved asynchronously, hence we need to wait for
+				// the entire prompt instruction tree to be processed
+				const instructionsStarted = performance.now();
+				await promptInstructions.allSettled();
+				// allow-any-unicode-next-line
+				this.logService.info(`[⏱] instructions tree resolved in ${performance.now() - instructionsStarted}ms`);
```

Comment:
This might be too much for info logs, I'd suggest using trace
```suggestion
				this.logService.trace(`[⏱] instructions tree resolved in ${performance.now() - instructionsStarted}ms`);
```